src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileIntersectionGraphBuilderImpl.java

```diff
@@ -0,0 +1,165 @@
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+
+/**
+ * Loads an NIH shapefile into an edge-based graph, associating the intersections with those previously
+ * imported from OpenStreetMap.
+ *
+ */
+public class NihShapefileIntersectionGraphBuilderImpl implements GraphBuilder {
+    private static Logger LOG = LoggerFactory.getLogger(ShapefileStreetGraphBuilderImpl.class);
+
+    private FeatureSourceFactory _featureSourceFactory;
+
+    public List<String> provides() {
+        // streets are required by the previous OSM graph builder step, so return an empty collection
+        return Collections.emptyList();
+    }
+
+    public List<String> getPrerequisites() {
+        // require the things provided by the OSM builder
+        return Arrays.asList("streets", "turns");
+    }
+
+    public void setFeatureSourceFactory(FeatureSourceFactory factory) {
+        _featureSourceFactory = factory;
+    }
+
+    public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
+        LOG.info("Adding to graph data from NIH Intersections Shapefile...");
+
+        FeatureSource<SimpleFeatureType, SimpleFeature> featureSource = _featureSourceFactory
+                .getFeatureSource();
+
+        FeatureCollection<SimpleFeatureType, SimpleFeature> features = null;
+
+        try {
+            CoordinateReferenceSystem sourceCRS = featureSource.getInfo().getCRS();
+
+            Hints hints = new Hints(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.TRUE);
+            CRSAuthorityFactory factory = ReferencingFactoryFinder.getCRSAuthorityFactory("EPSG",
+                    hints);
+            CoordinateReferenceSystem worldCRS = factory
+                    .createCoordinateReferenceSystem("EPSG:4326");
+
+            Query query = new Query();
+            query.setCoordinateSystem(sourceCRS);
+            query.setCoordinateSystemReproject(worldCRS);
+
+            features = featureSource.getFeatures(query);
+
+        } catch (Exception ex) {
+            throw new IllegalStateException("error loading shapefile intersection data", ex);
+        } finally {
+            _featureSourceFactory.cleanup();
+        }
+
+        // TODO: get rest of properties
+        StreetVertexIndexServiceImpl vertexSvc = new StreetVertexIndexServiceImpl(graph);
+        StringAttributeFeatureConverter intersectionIdSelector = new StringAttributeFeatureConverter("intsctn_id", "");
+
+        List<SimpleFeature> featureList = new ArrayList();
+        FeatureIterator<SimpleFeature> it2 = features.features();
+        while (it2.hasNext()) {
+            SimpleFeature feature = it2.next();
+            featureList.add(feature);
+        }
+        it2.close();
+        it2 = null;
+
+        LOG.info("Have {} NIH intersections to process...", featureList.size());
+
+        int foundIntersectionCt = 0;
+        int missingIntersectionCt = 0;
+
+        for (SimpleFeature feature : featureList) {
```

Comment:
I'm not sure why the original shapefile importer processes things that way, in two loops, apart from possibly isolating potential points of failure.  Yes, I think both importers could be handled in a single loop.